discovery-provider/src/tasks/generate_trending.py

```diff
@@ -16,209 +14,212 @@
 trending_cache_miss_key = 'trending_cache_miss'
 trending_cache_total_key = 'trending_cache_total'
 
-def generate_trending(db, time, genre, limit, offset):
-    identity_url = shared_config['discprov']['identity_service_url']
-    identity_trending_endpoint = urljoin(identity_url, f"/tracks/trending/{time}")
 
-    post_body = {}
-    post_body["limit"] = limit
-    post_body["offset"] = offset
+# Returns listens counts for tracks, subject to time and
+# genre restrictions.
+# Returns [{ track_id: number, listens: number }]
+def get_listen_counts(session, time, genre, limit, offset):
+
+    # Adds a created_at filter
+    # on the base query, if applicable.
+    def with_time_filter(base_query, time):
+        delta = None
+        if not time:
```

Comment:
in the case where no time is passed in, would the filter be where Play > current time?

Reply:
If no time is passed in, that means 'all time' so we don't add a time filter at all. `get_trending_tracks:L52`:
```
query_time = None if time not in ["day", "week", "month", "year"] else time
```

I can add a comment clarifying this behavior, it's not super clear

Reply:
oh i see, this returns base query right away. the rest set delta and then return after. these are two different if statements. this makes sense now